### PR TITLE
add prettier + ensure syntax workflow

### PR DESCRIPTION
**Pull request by OliverMKing**
# Description

This adds a `.prettierignore` and `.prettierrc` to enforce consistent formatting across the repository. This will automatically hook into most user's code formatters. You can learn more about Prettier from their [docs](https://prettier.io/).

A `.github/workflows/formatting.yml` file was also added to ensure that formatting was followed. It runs on any PRs.

The files mentioned here were the only ones with actual changes. You can view these changes specifically [here](https://github.com/Azure/aks-devx-tools/pull/53/commits/645c4db6dd9a9ac4b0e76f2a32224673d66bfc69). The rest of the modifications are simply formatting changes triggered to adhere to the Prettier configuration.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Passes all unit, integration tests and local testing. Workflow has been tested manually.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: SUPPORT.md [hunk 1]
```diff
@@ -1,25 +1,25 @@
...
-Support for this **PROJECT or PRODUCT** is limited to the resources listed above.
+# TODO: The maintainer of this repo has not yet edited this file
+
+**REPO OWNER**: Do you want Customer Service & Support (CSS) support for this product/project?
+
+-  **No CSS support:** Fill out this template with information about how to file issues and get help.
+-  **Yes CSS support:** Fill out an intake form at [aka.ms/onboardsupport](https://aka.ms/onboardsupport). CSS will work with/help you to determine next steps.
+-  **Not sure?** Fill out an intake as though the answer were "Yes". CSS will help you decide.
+
+_Then remove this first heading from this SUPPORT.MD file before publishing your repo._
```
[the remark this answers is not in the document]
Comment (OliverMKing) [reply SUPPORT.md:9]: Nothing changes.

If you view the rendered changes it says 

> No visible changes here. But you can view the diff anyway."

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,12 +2,12 @@
 
 ## 0.0.2
 
-* Improve docs links and details.
```
[the remark this answers is not in the document]
Comment (OliverMKing) [reply CHANGELOG.md:5]: That action follows the [keep a changelog](https://keepachangelog.com/en/1.0.0/) spec. The formatted version still follows the spec.

**Comment by davidgamero** [comment]
alrighty LGTM

**Review by davidgamero — APPROVED** [review]

**Comment by OliverMKing** [comment]
@Tatsinnit I could use your review on this when you get the chance 😄.

**Review by jaiveerk — APPROVED** [review]

**Review by bfoley13 — APPROVED** [review]
LGTM!

**Pull request merged**